_releases/2022-04-06-1.4.0-released.md

```diff
@@ -0,0 +1,82 @@
+
+Pre-built packages are available on [GitHub Releases](https://github.com/crystal-lang/crystal/releases/tag/1.4.0) and our official distribution channels.
+See [crystal-lang.org/install](https://crystal-lang.org/install/) for installation instructions.
+
+## Stats
+
+In this release we included [132 changes since the 1.3.2 release](https://github.com/crystal-lang/crystal/pulls?q=is%3Apr+milestone%3A1.4.0) by ?? contributors. We thank all the effort put into improving the language! ❤️
+
+## Towards WASM support
+
+Version 1.4.0 ships with, at the moment, minimal support for compiling into WebAssembly, linking it with a WASI-based LibC. There are several missing important pieces, check the [PR](https://github.com/crystal-lang/crystal/pull/10870). The updated instructions to try it out are:
+
+1. Write a simple Crystal program, let's say: 
+```crystal
+puts "Hello WebAssembly!"
+```
+and name it `main.cr`.
+You can also try more complicated programs, but keep in mind that this is still experimental
+and not ready for production.
+
+2. You need to have `wasm-ld` installed on your system, it is the WebAssembly linker from LLVM.
+On some systems, it comes by installing `lld`. Confirm that it is installed by using
+`wasm-ld --version`. Please also check that its version is similar to Crystal's LLVM version
+(`crystal --version`).
+
+3. You will need `libc` and `libpcre` compiled to the `wasm32-wasi` target. You can fetch a
+precompiled version of them [here](https://github.com/lbguilherme/wasm-libs/releases).
+Check for the `wasm32-wasi-libs.tar.gz` asset, download it and extract it.
+
+4. Cross-compile it with `crystal build main.cr --cross-compile --target wasm32-wasi`.
+
+5. Link it with
+```bash
+wasm-ld main.wasm -o main-final.wasm  -L "$PWD/wasm32-wasi-libs" -lc -lpcre -lclang_rt.builtins-wasm32
+```
+
+6. Run the WebAssembly module with `wasmer main-final.wasm` or `wasmtime main-final.wasm` and have fun.
+
+Note: There is currently a [PR](https://github.com/crystal-lang/crystal/pull/11948) that will simplify steps 4 and 5 above.
+
+## Better inference of instance and class variables
+
+Previous to this release, a simple program like the following failed to type because it couldn't infer the type of the instance variable:
+
+```crystal
+class DisplayHello
+  DELAY = 10.milliseconds
+
+  @timer_countdown = DELAY
+end
+```
+
+Now it compiles fine, inferring correctly that `@timer_countdown` has type `Time::Span`. It is also possible to omit types in other cases too ([details](https://github.com/crystal-lang/crystal/pull/11812)):
+
+```crystal
+class DisplayHello
+
+  def initialize(delay : Time::Span)
+    @timer_countdown = delay + 10.seconds
+  end
+end
+```
+
+
+## Other notable changes
+
+- Support for LLVM 14 ([PR](https://github.com/crystal-lang/crystal/pull/11905)).
+- Completed support for `Int128` ([PR](https://github.com/crystal-lang/crystal/pull/11576)).
```

Comment:
Well, apparently we're not fully complete yet. There are still a couple of things pending, including some new PRs.

Reply:
I'd say the compiler support is now probably finished, but there are still some holes in the stdlib.